src/transaction/log_manager.c

```diff
@@ -12471,6 +12471,144 @@ cdc_check_if_schema_changed (RECDES * recdes, HEAP_CACHE_ATTRINFO * attr_info)
   return or_rep_id (recdes) != attr_info->last_classrepr->id;
 }
 
+static int
+cdc_get_attribute_size (DB_VALUE * value)
+{
+  assert (!DB_IS_NULL (value));
+
+  int size = 0;
+
+  switch (DB_VALUE_TYPE (value))
+    {
+    case DB_TYPE_INTEGER:
+      size = sizeof (DB_C_INT);
+      break;
+    case DB_TYPE_BIGINT:
+      size = sizeof (DB_C_BIGINT);
+      break;
+    case DB_TYPE_SHORT:
+      size = sizeof (DB_C_SHORT);
+      break;
+    case DB_TYPE_FLOAT:
+      size = sizeof (DB_C_FLOAT);
+      break;
+    case DB_TYPE_DOUBLE:
+      size = sizeof (DB_C_DOUBLE);
+      break;
+    case DB_TYPE_NUMERIC:
+      size = DB_NUMERIC_BUF_SIZE;
+      break;
+    case DB_TYPE_BIT:
+    case DB_TYPE_VARBIT:
+      {
+	/* size of the data converted into bit string include "X''"
+	 * e.g. 17 = B'10001' = X'11' */
+	size = ((db_get_string_length (value) + 3) / 4) + 4;
```

Comment:
Could you tell me the meaning of `/4) + 4`? I think `(db_get_string_length (value) + 3)` means the string + `X''`, and?

Reply:
For the data `B'10001'`, `10001` is stored in DB_VALUE. 

db_get_string_length (value) = 5    //  number of 1, 0 in B'10001'

and when it convert value from B'10001' to X'11' (For reducing the size, it make binary to hex)  
((5 + 3) / 4 ) = 2           / number of 1, 0 in X'11'

And `(db_get_string_length (value) + 3) / 4) + 4` includes `\0`, so I will change it to 
`(db_get_string_length (value) + 3) / 4) + 3`

Reply:
Aha, 4 digits to 1 digit, OK, I got it.